api/src/test/java/com/epam/pipeline/controller/region/CloudRegionControllerTest.java

```diff
@@ -0,0 +1,333 @@
+
+        final ResponseResult<List<CloudProvider>> expectedResult =
+                ControllerTestUtils.buildExpectedResult(cloudProviders);
+
+        ControllerTestUtils.assertResponse(mvcResult, getObjectMapper(), expectedResult,
+                new TypeReference<Result<List<CloudProvider>>>() { });
+    }
+
+    @Test
+    public void shouldFailLoadAllForUnauthorizedUser() throws Exception {
+        mvc().perform(get(REGION_URL)
+                .servletPath(SERVLET_PATH))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAll() throws Exception {
+        final List<AwsRegion> cloudRegions = Collections.singletonList(awsRegion);
+
+        Mockito.doReturn(cloudRegions).when(mockCloudRegionApiService).loadAll();
+
+        final MvcResult mvcResult = mvc().perform(get(REGION_URL)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(EXPECTED_CONTENT_TYPE))
+                .andReturn();
+
+        Mockito.verify(mockCloudRegionApiService).loadAll();
+
+        final ResponseResult<List<AwsRegion>> expectedResult =
+                ControllerTestUtils.buildExpectedResult(cloudRegions);
+
+        ControllerTestUtils.assertResponse(mvcResult, getObjectMapper(), expectedResult,
+                new TypeReference<Result<List<AwsRegion>>>() { });
+    }
+
+    @Test
+    public void shouldFailLoadAllRegionsInfoForUnauthorizedUser() throws Exception {
+        mvc().perform(get(LOAD_REGIONS_INFO_URL)
+                .servletPath(SERVLET_PATH))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAllRegionsInfo() throws Exception {
+        final CloudRegionInfo awsRegionInfo = new CloudRegionInfo(awsRegion);
+        final List<CloudRegionInfo> cloudRegionInfos = Collections.singletonList(awsRegionInfo);
+
+        Mockito.doReturn(cloudRegionInfos).when(mockCloudRegionApiService).loadAllRegionsInfo();
+
+        final MvcResult mvcResult = mvc().perform(get(LOAD_REGIONS_INFO_URL)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(EXPECTED_CONTENT_TYPE))
+                .andReturn();
+
+        Mockito.verify(mockCloudRegionApiService).loadAllRegionsInfo();
+
+        final ResponseResult<List<CloudRegionInfo>> expectedResult =
+                ControllerTestUtils.buildExpectedResult(cloudRegionInfos);
+
+        final String actual= mvcResult.getResponse().getContentAsString();
+        Assert.assertTrue(StringUtils.isNotBlank(actual));
+        Assertions.assertThat(actual).isEqualToIgnoringWhitespace(getObjectMapper().writeValueAsString(expectedResult));
+
+        final Result<List<AbstractCloudRegion>> actualResult =
```

Comment:
Lets make the `CloudRegionInfo` a mutable object: add no args constructor, remove final and change `@Value` to `@Data`  to use `ControllerTestUtils#assertResponse` method